packages/nouns-subgraph/src/nouns-dao.ts

```diff
@@ -29,63 +32,69 @@ import {
   BIGINT_ZERO,
 } from './utils/constants';
 import { dynamicQuorumVotes } from './utils/dynamicQuorum';
+import { ParsedProposalV3, extractTitle } from './custom-types/ParsedProposalV3';
 
-/**
- * Extracts the title from a proposal's description. Returns 'Untitled' if the title is empty.
- * Note: Assemblyscript does not support regular expressions.
- * @param description The proposal description
- */
-function extractTitle(description: string): string {
-  // Extract a markdown title from a proposal body that uses the `# Title` or `Title\n===` formats
-  let splitDescription = description.split('#', 3);
-  if (splitDescription.length > 1) {
-    splitDescription.shift(); // Remove any characters before `#`
-  }
-  let title = splitDescription.join('').split('\n', 1).join('').trim();
-
-  // Remove bold and italics
-  title = title.replaceAll('**', '').replaceAll('__', '');
-
-  if (title == '') {
-    return 'Untitled';
-  }
-  return title;
+export function handleProposalCreatedWithRequirements(
+  event: ProposalCreatedWithRequirements1,
+): void {
+  handleParsedProposalV3(ParsedProposalV3.fromV1Event(event));
 }
 
-export function handleProposalCreatedWithRequirements(
+export function handleProposalCreatedWithRequirementsV3(
   event: ProposalCreatedWithRequirements,
 ): void {
-  let proposal = getOrCreateProposal(event.params.id.toString());
-  let proposer = getOrCreateDelegateWithNullOption(event.params.proposer.toHexString(), false);
+  handleParsedProposalV3(ParsedProposalV3.fromV3Event(event));
+}
+
+function handleParsedProposalV3(parsedProposal: ParsedProposalV3): void {
```

Comment:
@eladmallel maybe rename to something like `handleProposerCreated` ? `handleParsedProposalV3` is not very clear

Reply:
fixed!